### PR TITLE
Fixed exposure, and more

### PR DESCRIPTION
**Pull request by sagitta42**
!! Ignore the previous PR, is outdated, will need to be updated
Either leave it for now and I'll update later, or remove the PR altogether
This PR comes first

Main stuff:
- Fixed exposure to be calculated based on individual livetime of each detector
- `"event_type": "K_events"` works, allows to plot any parameter not only energy for K lines, works well as other event types. Didn't test thoroughly though
- cuts implemented but not tested yet gotta run will finish tomorrow -> so either merge this PR now to use exposure stuff, or wait until tomorrow for more and then merge, whatever works best, same thing for me

Small stuff:
Some small fixes here and there

### CODE REVIEW

**Comment by sagitta42** [comment]
Finished testing cuts, they work. Format like @sofia-calgaro original design: `"cuts": "is_valid_0vbb"` for "positive" cut (`==1`), or `"~is_valid_0vbb"` for "not" cut (`==0`). `"cuts"` can be a single cut or a list of cuts. Cuts are applied in `AnalysisData` after event selection and before other calculations such as mean, special parameters etc.

**Comment by sagitta42** [comment]
Tests were failing because the `cuts` module that is now deleted was imported in `__init__.py`. Removed.

**Comment by sagitta42** [comment]
Multi-parameter plotting now possible
- Json format `"parameters": ["A", "B"]`
- plot style available `"plot_style": "par vs par"`. 2D histo is WIP
- plotting range for multi-parameter possible as well, format `"range": {"A": [a0,a1], "B": [b0,b1]}`. Either parameter range is optional i.e. possible to provide range only for one parameter and skip the other one. Each point of range is optional e.g. can be `[null, a1]`
- in case of multi param per channel mode, mean is not in text; in case per string, fwhm is not in text - only plotted for single param case
- recap: for single parameter the format is `"range": [a,b]`. For single parameter plotting functions (i.e. all our previous functions except the new par vs par one), the following ones apply range where needed: vs time (on y axis ofc), histo (on x axis ofc)
Plots end up being heavy though, takes quite some time for the PDF to load after opened
Notes:
- the way it's implemented to propagate `plot_info` to either single-param or multi-param functions correctly is a bit messy. Implemented in such a way to preserve previous format (`plot_info["parameter"]` is a string, and so are unit, label etc) and allow multi-param format (`plot_info["parameters"]` contains list of params, unit and label are dicts with values for each parameter)
- tested that the single plotting did not get broken by this, worked
- should be possible for more than 2 params if that's somehow needed, tested only for 2 params e.g. A_max VS E

Other
- special parameter AoE_Custom = A_max / cuspEmax_ctc_cal
- small fixes

Warnings
- made some changes to saving == "append" case to simplify - please check if introduced errors!!
- made some changes to per cc4 function - sorry if breaks! tried to account

File: src/legend_data_monitor/plot_styles.py [hunk 1]
```diff
@@ -227,6 +249,10 @@ def plot_scatter(
         # edgecolors=color,
     )
 
+    if plot_info["K_events"]:
```
Comment (sofia-calgaro) [inline src/legend_data_monitor/plot_styles.py:252]: Is there a repetition? Already added at 226-229

File: src/legend_data_monitor/settings/remove-keys.json [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
-      "20230415T133659Z"
-    ],
-    "to": [
-      "20230401T012732Z",
-      "20230411T210547Z",
-      "20230413T084412Z",
-      "20230424T185631Z"
-    ]
-  }
+  "C00ANG3": [
```
Comment (sofia-calgaro) [inline src/legend_data_monitor/settings/remove-keys.json:2]: Wanted to do that initially, but wasn't sure about the effectiveness (also of the one I came up with). But ok, I hope maps will take care of these details in the future anyway :P

**Comment by sofia-calgaro** [comment]
...testing it before merge...

**Comment by sofia-calgaro** [comment]
Done following tests:

- new exposure function: previously tested, it works!
- append: **PROBLEMS!!!**. It crashes in ```analysis_data.py```
- cc4 function: it works! But wrong labels. They are of the type ```# - FWHM: X.XX``` where ```#``` I suppose being the channel. We need more info!
image
Eg see what's done in ```per string``` option:
image

- multiple parameters: ...not tested, considering the problems that are still there...
- one parameter range: it works!
	vs time -> works for: [null,null] - [null, B] (but it cuts lower values, why??) - [A, null] (but it cuts higher values, why??) - [A,B]
	histogram -> works for: [null,null] - [null, B] - [A, null] - [A,B]
- timestamp cuts for problematic detectors: they work! 
- quality cuts: applied correctly (tried with 1, and >1 flags)
- ```A/E_Custom```: it works!
- FWHM calculation, later displayed in plots: it works, but the rounding at 2 digits does not work for every parameter. We need more digits for some of them...


Notes: gonna apply the little fixes in the later PR I'll open, together with the ```append``` option.
Little fixes are: cc4 function labels, FWHM digits.

**Comment by sofia-calgaro** [comment]
[closing the PR was a mistake ugh]

File: src/legend_data_monitor/plot_styles.py (hunk 1 above)
Comment (sofia-calgaro) [reply src/legend_data_monitor/plot_styles.py:252]: ok, nevermind, from the web interface I didn't notice the two pieces belong to two different functions. There is no repetition:)

**Comment by sofia-calgaro** [comment]
Example of multiparameter config file [just for future reference]:

``` bash
"subsystems": {
    "geds": {
      "A/E versus Energy": {
        "parameters": ["cuspEmax_ctc_cal", "AoE_Custom"], \\ !!! - first param goes on x-axis, second one goes on y-axis
        "event_type": "pulser",
        "plot_structure": "per channel",
        "resampled": "no",
        "plot_style": "par vs par", \\ !!!
        "variation": false,
        "time_window": "1H"
      }
    }
```

**Pull request merged**